src/evaluate_queries.cpp

```diff
@@ -118,18 +122,32 @@ void evaluate_queries(const std::string &index_filename,
     auto source = std::make_shared<mio::mmap_source>(documents_filename.c_str());
     auto docmap = Payload_Vector<>::from(*source);
 
-    for (auto const & [ qid, query ] : enumerate(queries)) {
-        auto results = query_fun(query.terms);
+    std::vector<std::vector<std::pair<float, uint64_t>>> raw_results(queries.size());
+    auto start_batch = std::chrono::steady_clock::now();
+    tbb::parallel_for (size_t(0), queries.size(), [&] (size_t query_idx) {
+        auto qid = queries[query_idx].id;
+        raw_results[query_idx] = query_fun(queries[query_idx].terms);
+    });
+    auto end_batch = std::chrono::steady_clock::now();
+   
+    for (size_t query_idx = 0; query_idx < raw_results.size(); ++query_idx) { 
+        auto results = raw_results[query_idx];
+        auto qid = queries[query_idx].id;
         for (auto && [ rank, result ] : enumerate(results)) {
             std::cout << fmt::format("{}\t{}\t{}\t{}\t{}\t{}\n",
-                                     query.id.value_or(std::to_string(qid)),
+                                     qid.value_or(std::to_string(query_idx)),
                                      iteration,
                                      docmap[result.second],
                                      rank,
                                      result.first,
                                      run_id);
         }
     }
+    auto end_print = std::chrono::steady_clock::now();
+    double batch_ms = std::chrono::duration_cast<std::chrono::milliseconds>(end_batch - start_batch).count();
+    double batch_with_print_ms = std::chrono::duration_cast<std::chrono::milliseconds>(end_print - start_batch).count();
+    spdlog::info("Time taken to process queries: {}ms", batch_ms);
+    spdlog::info("Time taken to process queries with printing: {}ms", batch_with_print_ms);
```

Comment:
I would print this only in debug mode...

Reply:
Both printouts? Or just the printing one? I assumed we wanted an output from this - let me know what you are thinking and I can update.

Reply:
I would say just the second one

Reply:
Actually, let's differ this to a separate PR eventually...